packages/nextra/src/loader.ts

```diff
@@ -203,23 +191,22 @@ globalThis.__nextra_internal__ = {
   route: __nextra_pageOpts__.route,
 }
 
-function Content(props) {
+function Page(props) {
   return (
     <__nextra_SSGContext__.Provider value={props}>
-      <MDXContent />
+      {__nextra_withLayout__(
+        MDXContent,
+        {
+          titleText: typeof titleText === 'string' ? titleText : undefined,
+          ...__nextra_pageOpts__
+        },
+      ${layoutConfig ? '__nextra_layoutConfig__' : 'null'},
+      )}
     </__nextra_SSGContext__.Provider>
   )
 }
 
-export default __nextra_withLayout__(
-  ${JSON.stringify(pageNextRoute)},
-  Content,
-  {
-    titleText: typeof titleText === 'string' ? titleText : undefined,
-    ...__nextra_pageOpts__
-  },
-  ${layoutConfig ? '__nextra_layoutConfig__' : 'null'},
-)`.trimStart()
```

Comment:
Let me explain a bit on my previous change. _app.js renders the exported component from each page like `render(<Page/>)`. And because `function Page(props) {` is a different function on every different page, when a client navigation happens, React re-renders the `render(<Page2/>)`. Because `<Page/>` and `<Page2/>` have different types (`Page !== Page2`), React unmounts the `<Page/>` component and mounts the `<Page2/>`.

Because the layout stuff (navbar, sidebar, etc.) are all inside `<Page/>`, the states there will be reset too. That's why we had the `getLayout()` stuff.

And to avoid that, I refactored the `withLayout` HoC to always return the same component `Layout`, but use `route` to get the correct content to render inside. This way the layout will preserve the state and always be stable.

Reply:
> the states there will be reset too

There are two states, one is the React state (useState) and another one is the DOM state (scroll positions):

![CleanShot 2022-08-05 at 11 40 39](https://user-images.githubusercontent.com/3676859/183050568-dacaca4a-5a6e-4276-993a-f66ccbbb3986.gif)